src/App.js

```diff
@@ -2,24 +2,99 @@ import React from 'react';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+class App extends React.Component {
+  timerID = null;
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      date: new Date(),
+      isClockVisible: true,
+      clockName: 'clock',
+    };
+  }
+
+  componentDidMount() {
+    this.startClock();
+  }
+
+  componentWillUnmount() {
+    this.stopClock();
+  }
+
+  startClock = () => {
+    if (this.timerID === null) {
+      this.timerID = setInterval(() => {
+        this.tick();
+
+        // eslint-disable-next-line
+        console.log(this.state.date.toLocaleTimeString());
+      }, 1000);
+    }
+  }
+
+  stopClock = () => {
+    clearInterval(this.timerID);
+    this.timerID = null;
+  }
+
+  tick = () => {
+    this.setState({
+      date: new Date(),
+    });
+  }
+
+  showClock = () => {
+    this.startClock();
+    this.setState({
+      isClockVisible: true,
+    });
+  }
+
+  hideClock = () => {
+    this.stopClock();
+    this.setState({
+      isClockVisible: false,
+    });
+  }
+
+  setRandomName = () => {
+    const oldName = this.state.clockName;
+
+    this.setState({
+      clockName: `${Math.random()}`,
+    },
     // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+    () => console.log(`The Clock was renamed from ${oldName} to ${this.state.clockName}`));
+  }
+
+  render() {
+    return (
+      <>
+        <div className="App">
+          <h1>React clock</h1>
+          <p>
+            Current time:
+            {' '}
+            {this.state.isClockVisible && this.state.date.toLocaleTimeString()}
```

Comment:
```suggestion
            {this.state.isClockVisible && this.state.date.toLocaleTimeString()}
```
Let's read the task description more carefully. You should create a separate component `Clock`. It should have `name` and all lifecycle methods.
Also it should have `state`, which will contain `date: new Date(),`